server/server.js

```diff
@@ -61,20 +63,25 @@ const server = http.createServer((request, response) => {
             // Parse the post data and get client sent username and password.
             // let postDataObject = JSON.parse(postData);
             const page = DotCMSApi.processPage(JSON.parse(parse(postData).dotPageData).entity);
-            
+
             fs.readFile(`${STATIC_FOLDER}/index.html`, 'utf8', (err, data) => {
-                const app = renderToString(<App data={page.layout} />);
+                const context = {};
+                const app = renderToString(
+                    <StaticRouter location={request.url} context={context}>
+                        <Page data={page.layout} />
```

Comment:
why not put this into App?

Reply:
Becase `<App/>` now have a bunch of logic to handle the request to get get the page data and passing into `<Page/> in client mode but in server side we already have that data so we just pass it.